e2e/cypress/integration/enterprise/ldap_group/group_mentions_spec.js

```diff
@@ -0,0 +1,263 @@
+const saveConfig = (waitUntilConfigSaved = true) => {
+    // # Save if possible (if previous test ended abruptly all permissions may already be enabled)
+    cy.get('#saveSetting').then((btn) => {
+        if (!btn.disabled) {
+            btn.click();
+        }
+    });
+    if (waitUntilConfigSaved) {
+        cy.waitUntil(() => cy.get('#saveSetting').then((el) => {
+            return el[0].innerText === 'Save';
+        }));
+    }
+};
+
+describe('System Console', () => {
+    before(() => {
+        // # Login as sysadmin
+        cy.apiLogin('sysadmin');
+
+        // * Check if server has license for LDAP Groups
+        cy.requireLicenseForFeature('LDAPGroups');
+
+        // # Enable LDAP
+        cy.apiUpdateConfig({LdapSettings: {Enable: true}});
+
+        // # Link board group
+        cy.visit('/admin_console/user_management/groups');
+        cy.get('#board_group').then((el) => {
+            if (!el.text().includes('Edit')) {
+                // # Link the Group if its not linked before
+                if (el.find('.icon.fa-unlink').length > 0) {
+                    el.find('.icon.fa-unlink').click();
+                }
+            }
+        });
+
+        // # Get board group id
+        cy.apiGetGroups().then((res) => {
+            res.body.forEach((group) => {
+                if (group.display_name === 'board') {
+                    // # Set groupID to navigate to group page directly
+                    groupID = group.id;
+
+                    // # Set allow reference false to ensure correct data for test cases
+                    cy.apiPatchGroup(groupID, {allow_reference: false});
+                }
+            });
+        });
+
+        // # Login once as board user
+        boardUser = users['board-1'];
+        cy.apiLogin(boardUser.username, boardUser.password);
+
+        // # Login as sysadmin and add board-one to ad-1 team
+        cy.apiLogin('sysadmin');
+
+        // # Add board user to test team and to ad-1 to ensure that he has two teams so that we can see notifications on unopened team
+        cy.apiGetUserByEmail(boardUser.email).then((eRes) => {
+            const user = eRes.body;
+            cy.apiGetTeamByName('ad-1').then((teamRes) => {
+                cy.apiGetChannelByName('ad-1', 'town-square').then((channelRes) => {
+                    const channelId = channelRes.body.id;
+                    cy.apiAddUserToTeam(teamRes.body.id, user.id).then(() => {
+                        cy.apiAddUserToChannel(channelId, user.id);
+                    });
+                });
+            });
+        });
+    });
+
+    it('MM-23937 - Can enable and disable group mentions with a custom name for a group ', () => {
+        // # Login as sysadmin and navigate to board group page
+        navigateToGroup(groupID);
+
+        // # Click the allow reference button
+        cy.findByTestId('allow_reference_switch').then((el) => {
+            el.find('button').click();
```

Comment:
Click action to button works but it resets to original state (not active) when click action happened before the list of Users are loaded. My suggestion is to check Users first, something like:
```diff
+cy.findByText(boardUser.email).should('be.visible');
cy.findByTestId('allow_reference_switch').then((el) => {
    el.find('button').click();
});
```

Same case when disabling Group mentions.